reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/ScanWithSeed.kt

```diff
@@ -0,0 +1,40 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.completable.CompletableCallbacks
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.utils.lock.newLock
+import com.badoo.reaktive.utils.lock.synchronized
+
+fun <T, R> Observable<T>.scan(seed: R, accumulator: (acc: R, value: T) -> R): Observable<R> =
+    scan<T, R>({ seed }, accumulator)
+
+fun <T, R> Observable<T>.scan(getSeed: () -> R, accumulate: (acc: R, value: T) -> R): Observable<R> =
```

Comment:
Looks like we can avoid some boilerplate code here. Could you please try making `getSeed` lambda optional (nullable) and just call this method from the very first `scan` operator (which is without any `seed`)?

Reply:
I agree that it seems that there is some code duplication. The problem is that the upstream is of type `T` and down is `R`. If I drop the seed, then there is no way to transform `T` to `R`.

Reply:
Agree